packages/bsky/src/services/actor/views.ts

```diff
@@ -95,79 +85,100 @@ export class ActorViews {
+      .filter((list) => !!list)
+    const listViews = await this.services.graph.getListViews(listUris, viewer)
 
-    const views = results.map((result) => {
-      const profileInfo = profileInfoByDid[result.did]
-      const avatar = profileInfo?.avatarCid
+    return profileInfos.reduce((acc, cur) => {
+      const avatar = cur?.avatarCid
         ? this.imgUriBuilder.getCommonSignedUri(
             'avatar',
-            profileInfo.did,
-            profileInfo.avatarCid,
+            cur.did,
+            cur.avatarCid,
           )
         : undefined
-      const banner = profileInfo?.bannerCid
+      const banner = cur?.bannerCid
         ? this.imgUriBuilder.getCommonSignedUri(
             'banner',
-            profileInfo.did,
-            profileInfo.bannerCid,
+            cur.did,
+            cur.bannerCid,
           )
         : undefined
-      return {
-        did: result.did,
-        handle: result.handle ?? INVALID_HANDLE,
-        displayName: profileInfo?.displayName || undefined,
-        description: profileInfo?.description || undefined,
+      const profile = {
+        did: cur.did,
+        handle: cur.handle ?? INVALID_HANDLE,
+        displayName: cur?.displayName || undefined,
+        description: cur?.description || undefined,
         avatar,
         banner,
-        followsCount: profileInfo?.followsCount ?? 0,
-        followersCount: profileInfo?.followersCount ?? 0,
-        postsCount: profileInfo?.postsCount ?? 0,
-        indexedAt: profileInfo?.indexedAt || undefined,
+        followsCount: cur?.followsCount ?? 0,
+        followersCount: cur?.followersCount ?? 0,
+        postsCount: cur?.postsCount ?? 0,
+        indexedAt: cur?.indexedAt || undefined,
         viewer: viewer
           ? {
-              following: profileInfo?.requesterFollowing || undefined,
-              followedBy: profileInfo?.requesterFollowedBy || undefined,
-              muted: !!profileInfo?.requesterMuted || !!listMutes[result.did],
-              mutedByList: listMutes[result.did],
-              blockedBy: !!profileInfo.requesterBlockedBy,
-              blocking: profileInfo.requesterBlocking || undefined,
+              following: cur?.requesterFollowing || undefined,
+              followedBy: cur?.requesterFollowedBy || undefined,
+              muted: !!cur?.requesterMuted || !!cur.requesterMutedByList,
+              mutedByList: cur.requesterMutedByList
+                ? this.services.graph.formatListViewBasic(
+                    listViews[cur.requesterMutedByList],
+                  )
+                : undefined,
+              blockedBy: !!cur.requesterBlockedBy,
+              blocking: cur.requesterBlocking || undefined,
             }
           : undefined,
-        labels: labels[result.did] ?? [],
+        labels: labels[cur.did] ?? [],
       }
-    })
-
-    return Array.isArray(result) ? views : views[0]
+      return {
+        ...acc,
+        [cur.did]: profile,
+      }
```

Comment:
I generally don't focus too much on the style of accumulation (e.g. mutating vs immutable), but I wonder if we should actually consider mutating in hot paths like these view methods.  It does add-up to quite a few assignments and thrown-away objects that I think even today may not get optimized away by V8.

Reply:
In case it wasn't clear what I was getting at, e.g.:
```suggestion
      acc[cur.did] = profile
      return acc
```

Reply:
i'm game 👍